#  <p style="text-align: center;">Практическое задание к Семинару 3.<br><br>Урок 3. EDA (exploratory data analysis) или Разведочный анализ.</p>

In [2]:
from math import factorial, exp

In [3]:
import numpy as np

**Формула вероятности:**
<div class="alert alert-block alert-warning"><font size=4 color=black><b>
$\displaystyle P(A)=\frac{m}{n}$ (1)<br>, где $m$-число благоприятных исходов, $n$-общее число всех исходов.</b></font></div>

**Формула подсчета количества сочетаний по $k$ элементов из множества $n$:**  
<div class="alert alert-block alert-warning"><font size=4 color=black><b>$\displaystyle C^k_n=\frac{n!}{k!(n-k)!}$ (2)</b></font></div>

**Формула Бернулли:**
<div class="alert alert-block alert-warning"><font size=4 color=black><b>$\displaystyle P_{n}(X=k)=C^k_{n}p^{k}q^{n-k}$ (3)</b></font></div>

**Формула Пуассона:**
<div class="alert alert-block alert-warning"><font size=4 color=black><b>$\displaystyle P_m=\frac{\lambda^m}{m!}\cdot e^{-\lambda}$ (4)<br>, где $\lambda$ - среднее количество наступления события за определенную единицу измерений, и $\lambda=p\cdot n$.</b></font></div>

**Среднее арифметическое:**
<div class="alert alert-block alert-warning"><font size=4 color=black><b>$\displaystyle M(X)=\frac{1}{n}\sum^n_{i=1}x_i$ (5)</b></font></div>

**Среднее квадратичное отклонение (СКО) выборки:**
<div class="alert alert-block alert-warning"><font size=4 color=black><b>$\sigma=\sqrt{\frac{\displaystyle\sum^n_{i=1}(x_{i}-\bar x)^2}{n}}$ (6)</b></font></div>

**Смещенная дисперсия:**
<div class="alert alert-block alert-warning"><font size=4 color=black><b>$\sigma^2=\frac{\displaystyle\sum^n_{i=1}(x_{i}-\bar x)^2}{n}$ (7)</b></font></div>

**Несмещенная дисперсия:**
<div class="alert alert-block alert-warning"><font size=4 color=black><b>$\sigma^2=\frac{\displaystyle\sum^n_{i=1}(x_{i}-\bar x)^2}{n-1}$ (8)</b></font></div>

**Формула Баеса:**
<div class="alert alert-block alert-warning"><font size=4 color=black><b>$P_A(B_n)=\frac{P(B_n)\cdot P_{B_n}(A)}{P(A)}$ (9)</b></font></div>


In [4]:
def sochetaniy(n, k): #функция расчета количества сочетаний
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [5]:
def bernylli(n, k, p): # Формула Бернулли
    q = 1 - p;
    return float(sochetaniy(n, k)*p**k*q**(n-k))

In [6]:
def pyassona (m,p,n): # Формула Пуассона
    lambd=p*n
    return (lambd**m)/factorial(m)*exp(-lambd)

## ЗАДАНИЕ 1. 
**Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.**

**<p style="text-align: center;">
    <font size=10 color=#32CD32><b>-------------------------------</b></font><br>
    <font size=4.5 color=#3ac444>РЕШЕНИЕ:</font><br>
<font size=10 color=#32CD32><b>-------------------------------</b></font></p>**

<u>***Среднее арифметическое***</u>

In [7]:
ZP = [100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150]
total = 0      # или sum(ZP)
for i in ZP:
    total += i
mean_ar = total/len(ZP)
mean_ar                 # Средние арифметическое

65.3

<u>***Дисперсия***</u>

In [11]:
total_var = 0
for i in ZP:
    total_var += (i-mean_ar)**2
Variance_Bia = total_var/len(ZP) # Несмещенная дисперсия
Variance_Bia

950.11

In [12]:
Variance_Unb = total_var/(len(ZP)-1) # Cмещенная дисперсия
Variance_Unb

1000.1157894736842

<u>***Среднее квадратичное отклонение***</u>

In [13]:
CKO = np.sqrt(Variance_Bia) # Variance_Bia**0.5
CKO

30.823854398825596

<u>***ПРОВЕРКА С ПОМОЩЬЮ БИБЛИОТЕКИ NUMPY***</u>

In [14]:
ZP_np = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])
ZP_np.shape

(20,)

<u>***Среднее арифметическое***</u>

In [15]:
ZP_np.mean() #Среднее арифметическое

65.3

<u>***Дисперсия***</u>

In [16]:
np.var(ZP_np) # Несмещенная дисперсия

950.11

In [17]:
round(np.var(ZP_np, ddof=1),3) # Cмещенная дисперсия

1000.116

<u>***Среднее квадратичное отклонение***</u>

In [18]:
round(np.std(ZP_np),3)

30.824

<div class="alert alert-block alert-success"><font size=3 color='b'><b>ОТВЕТ: </b><br>Среднее арифметическое: М(Х) = 65.3<br>Cреднее квадратичное отклонение: $\sigma = 30.8239$<br>Несмещенная дисперсия: $\sigma^2_{\text{смещ}}=950.11$<br>Cмещенная дисперсия: $\sigma^2_{\text{несмещ}}=1000.116$

---

## ЗАДАНИЕ 2. 
**В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?**

**<p style="text-align: center;">
    <font size=10 color=#32CD32><b>-------------------------------</b></font><br>
    <font size=4.5 color=#3ac444>РЕШЕНИЕ:</font><br>
<font size=10 color=#32CD32><b>-------------------------------</b></font></p>**

Решаем по формуле 1, $\displaystyle P(A)=\frac{m}{n}$, т.к. из условия задачи имеются разные ящики, то вероятность будем рассчитывать для каждого ящика по отдельности. 
Общая вероятность будет находиться по формуле: $P(1,2,3)=P(1) + P(2) + P(3)$, для этого:<br>
**1 сценарий (P(1))** - Из первого ящика вынули 2 белых мяча, тогда из второго 1 белый мяч:<br>
1.1) Находим все благоприятные исходы в первом ящике (все сочетания белых мячей).  => $\displaystyle C^2_{5}$<br>
1.2) Находим все возможные исходы в первом ящике (все сочетания мячей).  => $\displaystyle C^2_{8}$<br>
1.3) Находим вероятность вытащить оба мяча подряд из первого ящика. => $P(A) = \frac {C^2_{5}} {C^2_{8}}$<br>
1.4) Находим все благоприятные исходы во втором ящике (вытащить 1 белый мяч).  => $\displaystyle C^1_{5}$<br>
1.5) Находим все благоприятные исходы во втором ящике (вытащить 3 НЕ белый мяч).  => $\displaystyle C^3_{7}$<br>
1.6) Находим все возможные исходы во втором ящике (все сочетания мячей).  => $\displaystyle C^4_{12}$<br>
1.7) Находим вероятность вытащить только 1 белый мяч из второго ящика. => $P(B) = \frac {C^1_{5}*C^3_{7}} {C^4_{12}}$<br>
1.8) Находим вероятность того, что 3 мяча белые. => $P(1_{(A+B)})=P(A)\cdot P(B)$

**2 сценарий (P(2))** - Из первого ящика вынули только 1 белый мяч и 1 НЕ белый мяч, тогда из второго ящика вынули 2 белых и 2 НЕ белых мяча:<br>
2.1) Находим все благоприятные исходы в первом ящике (все сочетания белых мячей).  => $\displaystyle C^1_{5}$<br>
2.2) Находим все благоприятные исходы в первом ящике (все сочетания НЕ белых мячей).  => $\displaystyle C^1_{3}$<br>
2.3) Находим все возможные исходы в первом ящике (все сочетания мячей).  => $\displaystyle C^2_{8}$<br>
2.4) Находим вероятность вытащить один белый и один НЕ белый мяч из первого ящика. => $P(A) = \frac {C^1_{5}+C^1_{3}} {C^2_{8}}$<br>
2.5) Находим все благоприятные исходы во втором ящике (2 белых мячя).  => $\displaystyle C^2_{5}$<br>
2.6) Находим все благоприятные исходы во втором ящике (сочетания 2 НЕ белых мячей).  => $\displaystyle C^2_{7}$<br>
2.7) Находим все возможные исходы во втором ящике (все сочетания мячей).  => $\displaystyle C^4_{12}$<br>
2.8) Находим вероятность вытащить два белый и два НЕ белый мяча из второго ящика. => $P(B) = \frac {C^2_{5}*C^2_{7}} {C^4_{12}}$<br>
2.9) Находим вероятность того, что все мячи белые. => $P(2_{(A+B)})=P(A)\cdot P(B)$

**3 сценарий (P(3))** - Из второго ящика вынули 3 белых мяча, тогда из первого 0 белых мячей:<br>
3.1) Находим все благоприятные исходы в первом ящике (все сочетания НЕ белых мячей).  => $\displaystyle C^2_{3}$<br>
3.2) Находим все возможные исходы в первом ящике (все сочетания мячей).  => $\displaystyle C^2_{8}$<br>
3.3) Находим вероятность вытащить оба НЕ белых мяча подряд из первого ящика. => $P(A) = \frac {C^2_{3}} {C^2_{8}}$<br>
3.4) Находим все благоприятные исходы во втором ящике (все сочетания белых мячей).  => $\displaystyle C^3_{5}$<br>
3.5) Находим все благоприятные исходы во втором ящике (все сочетания НЕ белых мячей).  => $\displaystyle C^1_{7}$<br>
3.5) Находим все возможные исходы во втором ящике (все сочетания мячей).  => $\displaystyle C^2_{12}$<br>
3.6) Находим вероятность вытащить 3 белых мяча и один НЕ белый мяч из второго ящика. => $P(B) = \frac {C^3_{5}*C^1_{7}} {C^4_{12}}$<br>
3.7) Находим вероятность того, что все мячи белые. => $P(3_{(A+B)})=P(A)\cdot P(B)$<br><br>

Находим вероятность того, что ровно два белых мяча:<br> 1 сценарий **ИЛИ** 2 сценарий **ИЛИ** 3 сценарий => **ИЛИ** = "+"<br>
P(1,2,3) = 1 сценарий **+** 2 сценарий **+** 3 сценарий


In [19]:
#Дано:
n_1 = 8
n_1_bel = 5
n_1_NO_bel = n_1 - n_1_bel

n_2 = 12
n_2_bel = 5
n_2_NO_bel = n_2 - n_2_bel

k_4 = 4
k_3 = 3
k_2 = 2
k_1 = 1

In [20]:
P_1 = sochetaniy(n_1_bel, k_2)/sochetaniy(n_1, k_2) * (sochetaniy(n_2_bel, k_1) * sochetaniy(n_2_NO_bel, k_3))/sochetaniy(n_2, k_4)
P_1

0.12626262626262627

In [21]:
P_2 = (sochetaniy(n_1_bel, k_1) * sochetaniy(n_1_NO_bel, k_1))/sochetaniy(n_1, k_2) * (sochetaniy(n_2_bel, k_2) * sochetaniy(n_2_NO_bel, k_2))/sochetaniy(n_2, k_4)
P_2

0.22727272727272727

In [22]:
P_3 = sochetaniy(n_1_NO_bel, k_2)/sochetaniy(n_1, k_2) * (sochetaniy(n_2_bel, k_3) * sochetaniy(n_2_NO_bel, k_1))/sochetaniy(n_2, k_4)
P_3

0.015151515151515152

In [23]:
P = P_1 + P_2 + P_3
P

0.3686868686868687

<div class="alert alert-block alert-success"><font size=3 color='b'><b>ОТВЕТ: </b><br>Вероятность того, что 3 мяча белые: $P\approx0.3687\approx36.87\%$</font></div>

---

## ЗАДАНИЕ 3. 
**На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен:<br>а) первым спортсменом?<br>б) вторым спортсменом?<br>в) третьим спортсменом?**

**<p style="text-align: center;">
    <font size=10 color=#32CD32><b>-------------------------------</b></font><br>
    <font size=4.5 color=#3ac444>РЕШЕНИЕ:</font><br>
<font size=10 color=#32CD32><b>-------------------------------</b></font></p>**

**Дано:**<br>
$A$ - в результате выстрела, цель поражена;  
$B_1$ - выстрел произведен первым спортсменом;  
$B_2$ - выстрел произведен вторым спортсменом;  
$B_3$ - выстрел произведен третьим спортсменом.
Т.к. выстрел, в результате которого цель будет поражена, может произвести любой из спортсменов, то эти события равновероятны и составляют полную группу событий. $\Rightarrow P(B_1)=P(B_2)=P(B_3)=\frac{1}{3}$ (1 попадание и 3 спортсмена).<br>
Вероятность того, что произошло событие $A$, при условии наступления события $B_n$ даны в условии задачи, т.е.<br>$P_{B_1}(A)=0.9,$<br>$P_{B_2}(A)=0.8,$<br>$P_{B_3}(A)=0.6$,<br>по формуле **полной** вероятности, вероятность наступления события:<br> $A$ - $P(A)=P(B_1)\cdot P_{B_1}(A)+P(B_2)\cdot P_{B_2}(A)+P(B_3)\cdot P_{B_3}(A)=\frac{1}{3}\cdot 0.9+\frac{1}{3}\cdot 0.8+\frac{1}{3}\cdot 0.6$  

In [39]:
P_B=1/3
P_A=P_B*0.9+P_B*0.8+P_B*0.6
print(f'Полная вероятность наступления события А Р(А) = {PA: .4f}')

Полная вероятность наступления события А Р(А) =  0.7667


Используя формулу **Байеса** (9), найдем оценку вероятности того, что могло произойти одно из событий $B_n$:
$P_A(B_1)=\frac{\frac{1}{3}\cdot 0.9}{0.7667}\\P_A(B_2)=\frac{\frac{1}{3}\cdot 0.8}{0.7667}\\P_A(B_1)=\frac{\frac{1}{3}\cdot 0.6}{0.7667}$  

In [41]:
P_AB_1=P_B*0.9/P_A
P_AB_2=P_B*0.8/P_A
P_AB_3=P_B*0.6/P_A
print(f'Вероятность того, что выстрел произвёл первый спортсмен: {P_AB_1: .4f};\n'
      f'Вероятность того, что выстрел произвёл второй спортсмен: {P_AB_2: .4f};\n'
      f'Вероятность того, что выстрел произвёл третий спортсмен: {P_AB_3: .4f}.'
     )

Вероятность того, что выстрел произвёл первый спортсмен:  0.3913;
Вероятность того, что выстрел произвёл второй спортсмен:  0.3478;
Вероятность того, что выстрел произвёл третий спортсмен:  0.2609.


<div class="alert alert-block alert-success"><font size=3 color='b'><b>ОТВЕТ: </b><br>а) Вероятность того, что выстрел произвёл первый спортсмен: $P_{A}(B_1)\approx0.3913\approx39.13\%$;<br>
б) Вероятность того, что выстрел произвёл второй спортсмен: $P_{A}(B_2)\approx0.3478\approx34.78\%$;<br> 
в) Вероятность того, что выстрел произвёл третий спортсмен: $P_{A}(B_3)\approx0.2609\approx26.09\%$.</font></div>

---

## ЗАДАНИЕ 4. 
**В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится:<br>а) на факультете A?<br>б) на факультете B?<br>в) на факультете C?**

**<p style="text-align: center;">
    <font size=10 color=#32CD32><b>-------------------------------</b></font><br>
    <font size=4.5 color=#3ac444>РЕШЕНИЕ:</font><br>
<font size=10 color=#32CD32><b>-------------------------------</b></font></p>**

**Дано:**<br>
$x$ - количество стедентов на факультете А и В;<br>
$2x$ - количество стедентов на факультете С;<br>
$Yra$ - сессия сдана;<br>
$Sdal_A$ - сессию сдал студент факультета A;<br>
$Sdal_B$ - сессию сдал студент факультета B;<br>
$Sdal_C$ - сессию сдал студент факультета C;<br>
Сдать первую сессию должен один из студентов факультетов, чтобы наступило событе Yra, тогда  $\Rightarrow P(Sdal_A)+P(Sdal_B)+P(Sdal_C)=1$ - образуют полную группу событий:  
$\frac{1}{n}+\frac{1}{n}+\frac{1}{2n}=1\Rightarrow n=2.5\\\Rightarrow P(Sdal_A)=\frac{1}{n}=0.25, P(Sdal_B)=\frac{1}{n}=0.25, P(Sdal_C)=\frac{1}{2n}=0.5$  
Вероятности наступления события "сессия сдана" ($Yra$), при условии что произошло одно из событий сессию сдал студент одного из факультетов $Sdal_n$ известны из условия задачи.  
Полная вероятность сдачи первой сессии студентом одного из факультетов исходя из формулы **полной** вероятности:  
$P(Yra)=0.25\cdot 0.8+0.25\cdot 0.7+0.5\cdot 0.9$

In [48]:
P_Yra=0.25*0.8+0.25*0.7+0.5*0.9
print(f'Полная вероятность сдачи сесссии студентом одного из факультетов, P(Yra) = {P_Yra}.')

Полная вероятность сдачи сесссии студентом одного из факультетов, P(Yra) = 0.825.


Используя формулу **Байеса** (9), найдем вероятность того, на каком факультете учиться студент сдавший сессию:<br>
$P_{𝑌𝑟𝑎}(Sdal_A)=\frac{P(SdalS_A)\cdot P_{Sdal_A}(𝑌𝑟𝑎)}{P(𝑌𝑟𝑎)}$ студент учится на факультете А  <br>
$P_{𝑌𝑟𝑎}(Sdal_B)=\frac{P(Sdal_B)\cdot P_{Sdal_B}(𝑌𝑟𝑎)}{P(𝑌𝑟𝑎)}$ студент учится на факультете В  <br>
$P_{𝑌𝑟𝑎}(Sdal_C)=\frac{P(Sdal_C)\cdot P_{Sdal_C}(𝑌𝑟𝑎)}{P(𝑌𝑟𝑎)}$ студент учится на факультете С

In [49]:
P_Yra_Sdal_A=0.25*0.8/P_Yra
P_Yra_Sdal_B=0.25*0.7/P_Yra
P_Yra_Sdal_C=0.5*0.9/P_Yra
print(f'Вероятность того, что студент учится на факультете А: {P_Yra_Sdal_A: .4f};\n'
      f'Вероятность того, что студент учится на факультете B: {P_Yra_Sdal_B: .4f};\n'
      f'Вероятность того, что студент учится на факультете C: {P_Yra_Sdal_C: .4f}.'
      )

Вероятность того, что студент учится на факультете А:  0.2424;
Вероятность того, что студент учится на факультете B:  0.2121;
Вероятность того, что студент учится на факультете C:  0.5455.


<div class="alert alert-block alert-success"><font size=3 color='b'><b>ОТВЕТ: </b><br>а) Вероятность того, что студент учится на факультете А:  $P_D(S_A)\approx0.2424\approx24.24\%$;<br>
б) Вероятность того, что студент учится на факультете B:  $P_D(S_B)\approx0.2121\approx21.21\%$;<br> 
в) Вероятность того, что студент учится на факультете C:  $P_D(S_C)\approx0.5455\approx54.55\%$.</font></div>

---

## ЗАДАНИЕ 5. 
**Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя:<br>а) все детали?<br>б) только две детали?<br>в) хотя бы одна деталь?<br>г) от одной до двух деталей?**

**<p style="text-align: center;">
    <font size=10 color=#32CD32><b>-------------------------------</b></font><br>
    <font size=4.5 color=#3ac444>РЕШЕНИЕ:</font><br>
<font size=10 color=#32CD32><b>-------------------------------</b></font></p>**

**Дано:**<br>
$A_1$ - в первый месяц вышла из строя 1 деталь;<br>
$A_2$ - в первый месяц вышла из строя 2 деталь;<br>
$A_2$ - в первый месяц вышла из строя 3 деталь;<br>

Вероятность выйти из строя в первый месяц:
- для **ПЕРВОЙ** детали: $P(A_1) = 0.1$
- для **ВТОРОЙ** детали: $P(A_2)=0.2$
- для **ТРЕТЬЕЙ** детали: $P(A_3)=0.25$

Противоположные события:<br>
$\bar A_1$ - в первый месяц НЕ вышла из строя 1 деталь;<br>
$\bar A_2$ - в первый месяц НЕ вышла из строя 2 деталь;<br>
$\bar A_3$ - в первый месяц НЕ вышла из строя 3 деталь;<br>

Вероятность **НЕ** выйти из строя в первый месяц:
- для **ПЕРВОЙ** детали:$P(\bar A_1)=1-P(A_1)=1-0.1=0,9$
- для **ВТОРОЙ** детали: $P(\bar A_2)=1-P(A_2)=1-0.2=0,8$
- для **ТРЕТЬЕЙ** детали: $P(\bar A_3)=1-P(A_3)=1-0.25=0,75$

**а)** Вероятность выйти из строя в первый месяц всех деталей, будет равна произведению вероятностей всех событий выхода из строя деталей:<br>$P(а)=P(A_1)\cdot P(A_2)\cdot P(A_3)=0.1\cdot 0.2\cdot 0.25$

In [65]:
P_a=0.1*0.2*0.25
print(f'Вероятность того, что в первый месяц из строя выйдут все детали Р(a) = {P_a: .4f}')

Вероятность того, что в первый месяц из строя выйдут все детали Р(a) =  0.0050


**б)** Вероятность выйти из строя в первый месяц тольку двух деталей, будет равна сумме вероятностей 3 сценариев:
- **1 сценарий:** В первый месяц вышли из строя 1 и 2 деталь  $\Rightarrow A_1$ **и** $A_2$ **и** $\bar A_3$;
- **2 сценарий:** В первый месяц вышли из строя 1 и 3 деталь  $\Rightarrow A_1$ **и** $A_3$ **и** $\bar A_2$;
- **3 сценарий:** В первый месяц вышли из строя 2 и 3 деталь  $\Rightarrow A_2$ **и** $A_3$ **и** $\bar A_1$;
$P(б)=P(A_1)\cdot P(A_2)\cdot P(\bar A_3)+P(A_1)\cdot P(A_3)\cdot P(\bar A_2)+P(A_2)\cdot P(A_3)\cdot P(\bar A_1)=0.1\cdot 0.2\cdot 0.75+0.1\cdot 0.25\cdot 0.8+0.2\cdot 0.25\cdot 0.9$

In [66]:
P_b=0.1*0.2*0.75+0.1*0.25*0.8+0.2*0.25*0.9
print(f'Вероятность того, что в первый месяц из строя только 2 детали Р(б) = {P_b: .4f}')

Вероятность того, что в первый месяц из строя только 2 детали Р(б) =  0.0800


**в)** Вероятность выйти из строя в первый месяц хотя бы одной деталь, можно вычислить через противоположное событие "не вышло из строя ни одной детали", тогда $P(в)$ вероятность события, что все детали работали в первый месяц, будет равна произведению вероятностей НЕ выхоода из строя в первый месяц всех трех деталей $\Rightarrow\bar A_1$, $\bar A_2$, и $\bar A_3$:<br> 
$P(Z)=P(\bar A_1)\cdot P(\bar A_2)\cdot P(\bar A_3)=0.9*0.8*0.75$

In [67]:
P_Z=0.9*0.8*0.75
print(f'Вероятность того, что из работать будут все детали равна Р(в) = {P_Z: .4f}')

Вероятность того, что из работать будут все детали равна Р(в) =  0.5400


Тогда вероятность выйти из строя хотя бы одной детали будет равно: $P(в)=1-P(Z)$

In [68]:
P_v=1-P_Z
print(f'Вероятность того, что выйдет из строя хотя бы одна деталь Р(в) = {P_v: .4f}')

Вероятность того, что выйдет из строя хотя бы одна деталь Р(в) =  0.4600


**г)** Вероятность выйти из строя в первый месяц от одной до двух деталей, будет равна сумме вероятностей 2 событий $\Rightarrow P(г)=P(1)+P(2)$, а именно:<br>1 событие - выйдет в первый из строя одна деталь;<br>2 событие - выйдет в первый из строя две детали;<br>
Вероятность 2-го события (выйдет в первый из строя две детали) нам известно из **б)**, тогда найдем вероятность P(1) 1-го события (выйдет в первый из строя одна деталь).<br>
**1 способ**<br>
Выйти из строя в первый месяц одна деталь может по трем сценариям:
- **1 сценарий:** В первый месяц вышли из строя 1 деталь  $\Rightarrow A_1$ **и** $\bar A_2$ **и** $\bar A_3$;
- **2 сценарий:** В первый месяц вышли из строя 2 деталь  $\Rightarrow A_2$ **и** $\bar A_1$ **и** $\bar A_3$;
- **3 сценарий:** В первый месяц вышли из строя 3 деталь  $\Rightarrow A_3$ **и** $\bar A_1$ **и** $\bar A_2$.
Вероятностью выхода в первый из строя одно детали  $P(1)$  будет равна **сумме** вероятностей этих сценариев:  
$P(1)=P(A_1)\cdot P(\bar A_2)\cdot P(\bar A_3)+P(A_2)\cdot P(\bar A_1)\cdot P(\bar A_3)+P(A_3)\cdot P(\bar A_1)\cdot P(\bar A_2)=0.1\cdot 0.8\cdot 0.75+0.2\cdot 0.9\cdot 0.75+0.25\cdot 0.9\cdot 0.8$  

In [69]:
P_1=0.1*0.8*0.75+0.2*0.9*0.75+0.25*0.9*0.8
print(f'Вероятность того, что выйдет из строя одна деталь Р(1) = {P_1: .4f}')

Вероятность того, что выйдет из строя одна деталь Р(1) =  0.3750


**2 способ**<br>Учитывая, что все варианты событий "не выйдет из строя ни одной детали", "из строя выйдет одна деталь", "из строя выйдет 2 детали", "из строя выйдут все детали", являются **несовместимыми**, и образуют полную группу событий, поэтому сумма вероятностей наступления этих событий равна $1$.  
$P(Z)+P(1)+P(б)+P(3а)=1\\0.54+P(1)+0.08+0.005=1\\P(1)=0.375$  

Найдйм вероятность выйти из строя в первый месяц от одной до двух деталей:<br>$P(г)=P(1)+P(б)=0.375+0.08=0.455$ 

<div class="alert alert-block alert-success"><font size=3 color='b'><b>ОТВЕТ: </b><br>а) Вероятность того что в первый месяц выйдут из строя все детали:  $P(3a)\approx0.005\approx0.50\%$;<br>
б) Вероятность того что в первый месяц выйдут из строя только 2 детали:  $P(б)\approx0.08\approx8.00\%$;<br> 
в) Вероятность того что в первый месяц выйдут из строя хотя бы одна деталь:  $P(в)\approx0.46\approx46.00\%$;<br>
г) Вероятность того что в первый месяц выйдут из строя от одной до двух деталей:  $P(г)\approx0.455\approx45.50\%$.</font></div>